cloud.foundry.cli/src/main/java/cloud/foundry/cli/operations/ServicesOperations.java

```diff
@@ -45,252 +48,241 @@ public ServicesOperations(DefaultCloudFoundryOperations cloudFoundryOperations)
         RenameServiceInstanceRequest renameServiceInstanceRequest = RenameServiceInstanceRequest.builder()
-            .name(currentName)
-            .newName(newName)
-            .build();
+                .name(currentName)
+                .newName(newName)
+                .build();
 
-        try {
-            this.cloudFoundryOperations.services()
+        //TODO: moove logs
+        return this.cloudFoundryOperations.services()
                 .renameInstance(renameServiceInstanceRequest)
-                .block();
-        } catch (RuntimeException e) {
-            throw new CreationException(e.getMessage());
-        }
+                .doOnSubscribe(aVoid -> {
+                    Log.debug("Rename service:", currentName);
+                    Log.debug("With new name:", newName);
+                });
     }
 
     /**
      * Update Tags, Plan of a Service Instance
      *
      * @param serviceInstanceName Name of a service instance
      * @param serviceBean         serves as template for the service to update
-     * @throws CreationException when the creation or the binding was not successful
+     * @return Mono which can be subscribed on to trigger the request to the cf instance
      */
-    public void updateServiceInstance(String serviceInstanceName, ServiceBean serviceBean) throws CreationException {
-        Log.debug("Update service Instance:", serviceInstanceName);
-        Log.debug("With the bean:", serviceBean);
-
+    public Mono<Void> update(String serviceInstanceName, ServiceBean serviceBean) {
+        //TODO:move logs
         UpdateServiceInstanceRequest updateServiceInstanceRequest = UpdateServiceInstanceRequest.builder()
-            .serviceInstanceName(serviceInstanceName)
-            .tags(serviceBean.getTags())
-            .planName(serviceBean.getPlan())
-            .build();
-
-        try {
-            this.cloudFoundryOperations.services()
-                    .updateInstance(updateServiceInstanceRequest)
-                    .block();
-
-            Log.info("Service Plan and Tags haven been updated");
-        } catch (RuntimeException e) {
-            throw new CreationException(e.getMessage());
-        }
+                .serviceInstanceName(serviceInstanceName)
+                .tags(serviceBean.getTags())
+                .planName(serviceBean.getPlan())
+                .build();
+                return this.cloudFoundryOperations.services()
+                        .updateInstance(updateServiceInstanceRequest)
+                        .doOnSubscribe(subscription -> {
+                            Log.debug("Update service Instance:", serviceInstanceName);
+                            Log.debug("With the bean:", serviceBean);
+                        });
     }
 
     /**
      * Deletes all keys and unbinds all routes and applications associated with the
-     * <code>serviceInstanceName</code>.
+     * <code>serviceInstanceName</code> and then deletes the service instance.
```

Comment:
Again we should update the documentation to clarify that this results in a request and that logging is performed within the Mono.